AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -560,6 +560,28 @@ - (void)displayImmediately
   [[self asyncLayer] displayImmediately];
 }
 
+- (void)__setNeedsLayout
+{
+  ASDisplayNodeAssertThreadAffinity(self);
+  ASDN::MutexLocker l(_propertyLock);
+  
+  if (!_flags.isMeasured) {
+    return;
```

Comment:
I agree that -setNeedsLayout should always result in a valid layout. But the main problem is that without a prior-measurement, the root node doesn't know its (soon-to-be-old) constrained size. It would be easily solved with a delegate, but as of now, this is the best we can do. (Another way is to can use an unconstrained size, but the calculated layout will be useless most of the time)